enso/utils.py

```diff
@@ -18,15 +20,22 @@ def feature_set_location(dataset_name, featurizer):
 
 def get_plugins(plugin_dir, match_names):
     """Responsible for grabbing objects from specified plugin directories."""
-    root, dirs, files = os.walk(plugin_dir).next()
+    full_plugin_dir_path = os.path.abspath(os.path.join(os.path.dirname(__file__), plugin_dir))
+    root, dirs, files = next(os.walk(full_plugin_dir_path))
+
     relevant_classes = []
     for filename in files:
-        if "__init__" not in filename and not filename.endswith('.pyc'):
-            module_name = filename.rpartition('.')[0]
-            mod = import_module("%s.%s" % (root, module_name))
-            for name, obj in inspect.getmembers(mod):
-                if inspect.isclass(obj) and obj.__name__ in match_names:
-                    relevant_classes.append(obj)
+        if filename.endswith('.pyc'):
```

Comment:
We should probably shift this logic to be affirmative rather than exclusionary.

Reply:
:+1:

Reply:
Revised implementation could be:

```
        if not filename.endswith('.py') or '__' in filename:
            continue
```